### PR TITLE
feat(growth): modifies guide assistant endpoint

### PR DESCRIPTION
**Pull request by roggenkemper**
this PR modifies the assistant endpoint so that guides can be restarted for the new sandbox walkthrough project. allows for the restart tour button in the end modal in https://github.com/getsentry/sentry/pull/40687 to work

### CODE REVIEW

File: src/sentry/api/endpoints/assistant.py [hunk 1]
```diff
@@ -81,16 +81,19 @@ def put(self, request: Request):
...
-        if status == "viewed":
-            fields["viewed_ts"] = timezone.now()
-        elif status == "dismissed":
-            fields["dismissed_ts"] = timezone.now()
-
-        try:
-            AssistantActivity.objects.create(user=request.user, guide_id=guide_id, **fields)
-        except IntegrityError:
-            pass
+        if status == "restart":
```
Comment (scefali) [inline src/sentry/api/endpoints/assistant.py:84]: @roggenkemper can you add a test for this?

File: src/sentry/api/endpoints/assistant.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ class AssistantEndpoint(Endpoint):
     permission_classes = (IsAuthenticated,)
 
     def get(self, request: Request) -> Response:
-        """Return all the guides with a 'seen' attribute if it has been 'viewed' or 'dismissed'."""
+        """Return all the guides with a 'seen' attribute if it has been 'viewed' o 'dismissed'."""
```
Comment (markstory) [inline src/sentry/api/endpoints/assistant.py:54]:
```suggestion
        """Return all the guides with a 'seen' attribute if it has been 'viewed' or 'dismissed'."""
```

File: src/sentry/api/endpoints/assistant.py (hunk 1 above)
Comment (markstory) [reply src/sentry/api/endpoints/assistant.py:84]: Should we have an enum for the status modes?

Comment (roggenkemper) [reply src/sentry/api/endpoints/assistant.py]: sounds like a good idea!

**Review by scefali — APPROVED** [review]

**Pull request merged**